### ISSUE TITLE
csrf analyzer not taking into acount [FromBody] decoracted input

### ISSUE THREAD

**Issue by justinstanley** [security-code-scan/security-code-scan#83]
The csrf analyzer should also consider the [FromBody] attribute on the controller action input parameters. 
[FromBody] indicates the data to be POSTed by the client is JSON, hence not susceptible to XSRF.

We could certainly add an suppression comment here, but I feel the logic should be considering the input decorated with [FromBody] as valid. 

```cs
[HttpPost]
[IgnoreAntiforgeryToken(Order = 1000)]
public JsonResult MyControllerAction([FromBody] Foo model)
{
	var bar = new Bar();
	return Json(bar)
}
```

**Comment by JarLob** [comment]
Hi, sorry for late reply. I think it is a correct assumption in 99% occurrences. If the [endpoint accepts `text/plain`](https://stackoverflow.com/questions/11008469/are-json-web-services-vulnerable-to-csrf-attacks) the request still can be forged from a form. However [it has to be explicitly enabled](https://stackoverflow.com/questions/25631970/how-to-post-plain-text-to-asp-net-web-api-endpoint). Thus the warning on such endpoints can be disabled by default, but it would be good to still have a warning in `AuditMode` for review.

**Comment by JarLob** [comment]
Closed via 8eb60fb45951db15d4e8a31628013acede318dc7
